pkg/controller/controller_multiplex.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog/v1beta1"
+	"github.com/kubernetes-incubator/service-catalog/pkg/pretty"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// Conditions
+
+// ConditionType can be translated to a ServiceInstanceCondition or
+// ServiceBindingCondition value.
+type ConditionType string
+
+const (
+	// ConditionReady represents that a given InstanceCondition is in
+	// ready state.
+	ConditionReady ConditionType = "Ready"
+
+	// ConditionFailed represents information about a final failure
+	// that should not be retried.
+	ConditionFailed ConditionType = "Failed"
+)
+
+// setCondition sets a single condition on a type's status. This delegates to
+// the correct type implementation.
+//
+// Note: objects coming from informers should never be mutated; always pass a
+// deep copy as the instance parameter.
```

Comment:
is this note accurate?

Are we handling the deepcopy around the actual uses? 

Can we handle the deepcopy inside the method instead, or are we expecting that we're mutating what's passed in rather than passing out the modified object?

Reply:
I did not write the code or the comment, just moved it. Someone else will have to speak to those questions.

Reply:
Okay, this looks copied and not moved. Are there still more instances of `setServiceBindingCondition` and `setServiceInstanceCondition` that need to be converted?

Reply:
I am not focused on migrating `set*Condition` functions in this PR, more I wanted to have `prepareInProgressProperties` to be generic but to be able to do that I needed a couple generic helpers like `setCondition` and `updateStatus`

Reply:
RE passing in deep copies - we already do that. we should explore in another PR whether we can move the deep-copying into the functions.